base/dict.jl

```diff
@@ -68,12 +68,80 @@ Dict{String,Int64} with 2 entries:
 
 Alternatively, a sequence of pair arguments may be passed.
 
-```jldoctest
-julia> Dict("A"=>1, "B"=>2)
+```jldoctest dictexamples
+julia> d = Dict("A"=>1, "B"=>2)
 Dict{String,Int64} with 2 entries:
   "B" => 2
   "A" => 1
 ```
+
+After creation of a Dict you can use `d[x]` for getting the values and `d[x]=y` for changing the values.
+```jldoctest dictexamples
+julia> d["A"]
+1
+
+julia> d["B"] = 3
+3
+```
+
+# Extended help
+
+You can use [Generator](https://docs.julialang.org/en/v1/manual/arrays/#Generator-Expressions-1) and [Comprehension][https://docs.julialang.org/en/v1/manual/arrays/#Comprehensions-1] syntax to create dictionaries:
+
+```jldoctest
+julia> Dict(x => x^2 for x = 0:1:4)
+Dict{Int64,Int64} with 5 entries:
+  0 => 0
+  4 => 16
+  2 => 4
+  3 => 9
+  1 => 1
+```
+
+It is usally possible to collect the keys and values using arrays in a for-loop and then create a `Dict`.
+However, you may want to create and fill dictionaries dynamically (slower than the other methods).
+
+First, create an empty dictionary using:
+```jldoctest
+julia> Dict{Int64,Any}()  # type specified
+Dict{Int64,Any} with 0 entries
+
+julia> Dict()  # type not specified
+Dict{Any,Any} with 0 entries
+```
+
+After creating an empty dictionary, fill it via a for loop:
```

Comment:
This is the doctstring for the constructor `Dict()`. Now we are talking about modyfing an existing dict which feels out of scope for this docstring.

Reply:
This is in the Extended help section of `Dict`. `Dict` doesn't have a manual like [`Array` does](https://docs.julialang.org/en/v1/manual/arrays/), so this docstring is the best place for talking about modifying a Dict.